packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -50,26 +53,46 @@ export default function createVitePluginAstroServer({
 			const controller = createController({ loader });
 			const localStorage = new AsyncLocalStorage();
 
-			/** rebuild the route cache + manifest, as needed. */
-			async function rebuildManifest(needsManifestRebuild: boolean) {
+			/** rebuild the route cache + manifest */
+			async function rebuildManifest(path: string | null = null) {
 				pipeline.clearRouteCache();
-				if (needsManifestRebuild) {
+
+				// If a route changes, we check if it's part of the manifest and check for its prerender value
+				if (path !== null) {
+					const route = routeManifest.routes.find(
+						(r) => path === new URL(r.component, settings.config.root).pathname,
```

Comment:
I don't think we need to use `new URL`. I `path.endsWith(r.component)` should be enough, because `r.component` is already a byproduct of `URL`